From 6d0e72072bb735afedd507307a312018e40bd6ba Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Fri, 9 Dec 2016 15:32:46 +0800
Subject: [PATCH 1345/5242] MLK-13570-8 usb: chpidea: usbmisc_imx: add
 non-burst setting for both imx7d and imx7ulp

commit  5c630c6abe505b983fe23c66599d853a8ee6cd65 from
https://source.codeaurora.org/external/imx/linux-imx.git

For all imx Socs later than imx6 (including imx6), the USB_nSBUSCFG.AHBBRST
will be set as 0 at dtsi file, so the non-burst setting needs to be
set at non-core register, or there will be no burst for USB AHB/AXI
transfer.

Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/usbmisc_imx.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/usb/chipidea/usbmisc_imx.c b/drivers/usb/chipidea/usbmisc_imx.c
index f8a1102..78fbacf 100644
--- a/drivers/usb/chipidea/usbmisc_imx.c
+++ b/drivers/usb/chipidea/usbmisc_imx.c
@@ -601,6 +601,10 @@ static int usbmisc_imx7d_init(struct imx_usbmisc_data *data)
 	}
 	writel(reg, usbmisc->base);
 
+	/* SoC non-burst setting */
+	reg = readl(usbmisc->base);
+	writel(reg | MX6_BM_NON_BURST_SETTING, usbmisc->base);
+
 	reg = readl(usbmisc->base + MX7D_USBNC_USB_CTRL2);
 	reg &= ~MX7D_USB_VBUS_WAKEUP_SOURCE_MASK;
 	writel(reg | MX7D_USB_VBUS_WAKEUP_SOURCE_BVALID,
-- 
1.7.9.5

